### ISSUE TITLE
Intellisense issues in VSCode

### ISSUE THREAD

**Issue by aaronschiff** [quarto-dev/quarto#190]

**Comment by jjallaire** [comment]
This might be a conflict with an other extension that is mutating the "background" document we use for doing completions? Another possibility is that you have auto-save enabled and this is confounding what we do. In either case I'm sure we can find our way through this. So a couple of questions:

(1) Do you have autosave enabled?

(2) Are there any other extensions you have that have anything to do with saving or formatting files?

(3) If there is nothing obvious for (2), try executing the **Help: Start Extension Bisect** command. This will allow you to narrow down if there is another extension which we are interacting with undesirably (it will basically progressively enable/disable extensions via binary search and ask you to try repoing the error in every state).

**Comment by aaronschiff** [comment]
@jjallaire thanks for your help. I have autosave disabled, and have also tried turning off the Format On Save and Format On Type options but still get the error popping up. I will work through the process with other extensions and report back.

**Comment by aaronschiff** [comment]
Another observation - frequently when editing a qmd document I see this information dialog pop up. It seems to run for ~10 seconds every few minutes (this is with a ~2000 lines qmd). The intellisense.r file in a temporary directory also pops open in the VS Code editor. 

<img width="460" alt="Screenshot 2023-06-01 at 11 40 44" src="https://github.com/quarto-dev/quarto/assets/10150126/9414d0c3-f7e2-4630-84d5-ee4afae66825">

**Comment by jjallaire** [comment]
Do you know if the R extension has a special comment that can be used to disable the auto-formatter? I'm pretty sure that this long formatting time is the source of the issue so if we could disable it that would be a good solution.

**Comment by aaronschiff** [comment]
Yes it is possible to disable R extension's formatting with `r.editor.formatOnType: false` and/or `r.editor.formatOnSave: false` in VSCode settings. That does seem to resolve the issue I'm seeing with the intellisense.r file but also disables formatting of other .r files so maybe not an ideal solution if working with both .r and .qmd files at the same time in a workspace.

**Comment by jjallaire** [comment]
I wonder if there is a way to disable it on a per-file basis? (e.g. through a special comment)

**Comment by aaronschiff** [comment]
Might be possible. I think REditorSupport uses styler for formatting code and the styler [documentation](https://styler.r-lib.org/dev/reference/stylerignore.html) explains how to turn styling on and off via comments. I'm not sure how to apply this to the temporary .intellisense.r file though.  

Also just to clarify it is only `r.editor.formatOnSave: false` that seems to fix the issue with slow formatting of .intellisense.r globally, although a per-file solution would be better.

**Comment by jjallaire** [comment]
I think we could add the comment within the extension (we do the same for Python linters).

**Comment by jjallaire** [comment]
This commit should do it: https://github.com/quarto-dev/quarto/commit/ec897f43d3487856676bfed17a56573872ad31ea

We'll have a new release in the next few days that will have the fix.

**Comment by aaronschiff** [comment]
Amazing, thanks!
